File: src/cpp/explore/topquestions/easy/array/加一.cpp
#include <bits/stdc++.h>
using namespace std;

/**
 * @File    :   加一.cpp
 * @Time    :   2020/07/25 12:03:55
 * @Author  :   wylu
 * @Version :   1.0
 * @Contact :   15wylu@gmail.com
 * @License :   Copyright © 2020, wylu-CHINA-SHENZHEN. All rights reserved.
 * @Desc    :   https://leetcode-cn.com/explore/featured/card/top-interview-questions-easy/1/array/27/
 */
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int carry = 1;
        for (int i = digits.size() - 1; i >= 0 && carry != 0; i--) {
            carry += digits[i];
            digits[i] = carry % 10;
            carry /= 10;
        }

        if (carry) {
            digits.insert(digits.begin(), carry);
        }
        return digits;
    }
};

File: src/python/explore/topquestions/easy/array/加一.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
@File    :   加一.py
@Time    :   2020/07/26 10:10:28
@Author  :   wylu
@Version :   1.0
@Contact :   15wylu@gmail.com
@License :   Copyright © 2020, wylu-CHINA-SHENZHEN. All rights reserved.
@Desc    :
"""
from typing import List


class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        carry = 1
        for i in range(len(digits) - 1, -1, -1):
            carry += digits[i]
            digits[i] = carry % 10
            carry //= 10

        if carry > 0:
            digits.insert(0, carry)

        return digits
